.github/workflows/code-quality.yml

```diff
@@ -102,21 +106,67 @@ jobs:
         id: meta
         run: |
           echo "sha_short=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+          echo "pr_number=${{ github.event.pull_request.number }}" >> $GITHUB_OUTPUT
+          echo "github_ref=${{ github.ref }}" >> $GITHUB_OUTPUT
+          if [ "${{ github.event_name }}" = "push" ] && [ "${{ github.ref }}" = "refs/heads/main" ]; then
+            echo "is_main_push=true" >> $GITHUB_OUTPUT
+          else
+            echo "is_main_push=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Login to Azure Container Registry
+        if: steps.meta.outputs.is_main_push == 'true'
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ secrets.AZURE_CLIENT_ID }}
+          password: ${{ secrets.AZURE_CLIENT_SECRET }}
```

Comment:
Recommend to use  az login github action, and then `az acr login ` command here instead of this - using the creds secret you have set up